lib/intl.ts

```diff
@@ -113,10 +166,21 @@ export function DateTimeFormat(
   this[DATETIME] = datetimeAmend;
   this[ZONED] = zonedDateTimeAmend;
   this[INST] = instantAmend;
+  return undefined; // TODO: I couldn't satisfy TS without adding this. Is there another way?
```

Comment:
This is a bit worrying... is the intent that DateTimeFormatImpl be a private impl of the DateTimeFormat ctor?

Reply:
Yes. Typing this damn function was a huge hassle. See https://github.com/js-temporal/temporal-polyfill/pull/81#discussion_r726540190. LMK if you can think of a better solution.